hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/logic/ThrottleException.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.workflows.handle.flow.dispatch.logic;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.ResponseCodeEnum;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * This class is used to throw a {@link ThrottleException} when a transaction is gas throttled.
+ */
+public class ThrottleException extends Exception {
```

Comment:
Exception throwing can be expensive (depending on whether stack traces are being filled in or not). Normally you would not use exceptions for control flow. In this particular case, I am not sure we can say that hitting a throttle is an exceptional situation -- it can happen quite easily on the backend because we cannot do gas filtering on ingest. I haven't looked into the call sites for this, but I am concerned over using an exception for "when a transaction is gas throttled".

Reply:
We had to distinguish the kind of `HandleException` that was happening [here](https://github.com/hashgraph/hedera-services/blob/release/0.51/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java#L594) to charge different fees. In the new code is caught [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/logic/DispatchProcessor.java#L154) . Do you recommend any other approach?